src/Liquidator.sol

```diff
@@ -274,33 +275,43 @@ contract Liquidator is Owned, ReentrancyGuard, ILiquidator {
     /**
      * @notice Places a bid.
      * @param account The contract address of the Account being liquidated.
-     * @param askedAssetAmounts Array with the assets-amounts the bidder wants to buy.
+     * @param bidAmounts Array with the assets-amounts the bidder wants to buy.
      * @param endAuction_ Bool indicating that the auction can be ended after the bid.
+     * @param data Data passed through back to the bidder via the bidCallback() call.
      * @dev We use a Dutch auction: price of the assets constantly decreases.
-     * @dev The "askedAssetAmounts" array should have equal length as the stored "assetAmounts" array.
+     * @dev The "bidAmounts" array should have equal length as the stored "assetAmounts" array.
      * An amount 0 should be passed for assets the bidder does not want to buy.
+     * If the length of "bidAmounts" does not equal "assetAmounts", the function will revert in the Registry.
      * @dev The bidder is not obliged to set endAuction to True if the account is healthy after the bid,
      * but they are incentivised to do so by earning an additional "auctionTerminationReward".
+     * @dev If the bidder is a contract, it must implement the IBidCallback interface.
      */
-    function bid(address account, uint256[] memory askedAssetAmounts, bool endAuction_) external nonReentrant {
+    function bid(address account, uint256[] memory bidAmounts, bool endAuction_, bytes calldata data)
+        external
+        nonReentrant
+    {
         AuctionInformation storage auctionInformation_ = auctionInformation[account];
         if (!auctionInformation_.inAuction) revert LiquidatorErrors.NotForSale();
 
-        // Calculate the current auction price of the assets being bought.
-        uint256 totalShare = _calculateTotalShare(auctionInformation_, askedAssetAmounts);
+        // Transfer the assets to the bidder, returns updated bidAmounts with the actual bought assets.
+        // The actual bought assets are equal or smaller than the asked assets.
+        bidAmounts = IAccount(account).auctionBid(
+            auctionInformation_.assetAddresses, auctionInformation_.assetIds, bidAmounts, msg.sender
+        );
+
+        // Calculate the current auction price of the assets bought.
+        uint256 totalShare = _calculateTotalShare(auctionInformation_, bidAmounts);
         uint256 price = _calculateBidPrice(auctionInformation_, totalShare);
 
+        // Hook back to the bidder with the actual amounts of bought assets and the actual bid price.
+        // Bidders can optionally first liquidate the assets before repaying the debt.
```

Comment:
```suggestion
        // Bidders can optionally first liquidate the assets before repaying the debt.
        // Bidders should NOT call a LendingPool.repay() or LendingPool.auctionRepay(), as this will cause the bidder to pay twice!
```